### ISSUE TITLE
Move LogitGetter functionality into loss functions

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#256]
LogitGetter should be deprecated and the inference logic should be moved to the classifier loss functions.
